### ISSUE TITLE
AndroidClientHandler uses wrong base class on .NET6

### ISSUE THREAD

**Issue by marek-safar** [xamarin/xamarin-android#5754]
[AndroidClientHandler](https://github.com/xamarin/xamarin-android/blob/main/src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs) implementation derives from `HttpClientHandler` which is wrong on .NET6 because `HttpClientHandler` **always** uses `SocketHttpHandler` as underlying implementation. This will cause problems because some properties/method will read/write data using `SocketsHttpHandler` and the rest will use `Java.Net.HttpURLConnection`

The correct type hierarchy is to derive from `System.Net.Http.HttpMessageHandler`.

@steveisok @pjcollins @jonathanpeppers

**Comment by jonathanpeppers** [comment]
I was checking on this to see if we could just change the base class, and we actually *use* several of the members present on `HttpClientHandler`:

```
AndroidClientHandler.cs(268,9): error CS0103: The name 'UseProxy' does not exist in the current context
AndroidClientHandler.cs(315,31): error CS0119: 'Proxy' is a type, which is not valid in the given context
AndroidClientHandler.cs(319,21): error CS0117: 'Proxy' does not contain a definition for 'GetProxy'
AndroidClientHandler.cs(573,9): error CS0103: The name 'AllowAutoRedirect' does not exist in the current context
AndroidClientHandler.cs(636,41): error CS0103: The name 'MaxAutomaticRedirections' does not exist in the current context
AndroidClientHandler.cs(637,81): error CS0103: The name 'MaxAutomaticRedirections' does not exist in the current context
AndroidClientHandler.cs(722,9): error CS0103: The name 'UseCookies' does not exist in the current context
AndroidClientHandler.cs(722,23): error CS0119: 'CookieContainer' is a type, which is not valid in the given context
AndroidClientHandler.cs(731,5): error CS0120: An object reference is required for the non-static field, method, or property 'CookieContainer.SetCookies(Uri, string)'
AndroidClientHandler.cs(722,117): error CS0165: Use of unassigned local variable 'cookieHeaderValue'
AndroidClientHandler.cs(854,9): error CS0103: The name 'AutomaticDecompression' does not exist in the current context
AndroidClientHandler.cs(859,9): error CS0103: The name 'AutomaticDecompression' does not exist in the current context
AndroidClientHandler.cs(864,8): error CS0103: The name 'AutomaticDecompression' does not exist in the current context
AndroidClientHandler.cs(872,8): error CS0103: The name 'UseCookies' does not exist in the current context
AndroidClientHandler.cs(872,22): error CS0119: 'CookieContainer' is a type, which is not valid in the given context
AndroidClientHandler.cs(873,32): error CS0120: An object reference is required for the non-static field, method, or property 'CookieContainer.GetCookieHeader(Uri)'
AndroidClientHandler.cs(953,9): error CS0103: The name 'PreAuthenticate' does not exist in the current context
AndroidClientHandler.cs(956,46): error CS0119: 'Proxy' is a type, which is not valid in the given context
AndroidClientHandler.cs(956,67): error CS0103: The name 'Credentials' does not exist in the current context
```

If we override `SendAsync()` completely and do our own thing, is there actually a problem here?

**Comment by steveisok** [comment]
@jonathanpeppers What I think needs to happen is that there needs to be another type that derives from `HttpMessageHandler`, move a bunch of the logic into there, and have `AndroidClientHandler` call into it.

**Comment by steveisok** [comment]
Also, the description of this issue is a bit different than what we will have in runtime.  `HttpClientHandler` will choose whether to use the native Android handler or `SocketsHttpHandler` and its underlying handler.

**Comment by grendello** [comment]
This was fixed in 1e5bfa3ff0ae18ce9407677667f05a9bfba53478
